src/Zigzag_persistence/include/gudhi/filtered_zigzag_persistence.h

```diff
@@ -0,0 +1,470 @@
+                return p1.first < p2.first;
+              });
+      if (itBirth == filtrationValues_.end() || itBirth->first > birthKey) {
+        --itBirth;
+      }
+      return itBirth->second;
+    };
+
+    auto stream_infinite_interval = [&](dimension_type dim, internal_key birthIndex) {
+      if (dimMax_ == -1 || (dimMax_ != -1 && dim < dimMax_))
+        diag.emplace_back(birth(birthIndex), Filtration_value_interval::inf, dim);
+    };
+
+    pers_.get_current_infinite_intervals(stream_infinite_interval);
+  }
+};  // end class Filtered_zigzag_persistence_with_storage
+
+/**
+ * @ingroup zigzag_persistence
+ *
+ * @brief Class computing the zigzag persistent homology of a zigzag filtration. Algorithm based on \cite zigzag.
+ * 
+ * @tparam FilteredZigzagOptions Structure following the @ref FilteredZigzagOptions concept.
+ * Default value: @ref Default_filtered_zigzag_options.
+ */
+template <class FilteredZigzagOptions = Default_filtered_zigzag_options>
+class Filtered_zigzag_persistence {
+ public:
+  using Options = FilteredZigzagOptions;                       /**< Zigzag options. */
+  using internal_key = typename Options::internal_key;         /**< Key and index type, has to be signed. */
+  using face_key = typename Options::face_key;                 /**< Face ID type from external inputs. */
+  using filtration_value = typename Options::filtration_value; /**< Type for filtration values. */
+  using dimension_type = typename Options::dimension_type;     /**< Type for dimension values. */
+
+  /**
+   * @brief Constructor.
+   * @details After construction of the class, the zigzag filtration should be given in a streaming like way, i.e.,
+   * call @ref insert_face, @ref remove_face or @ref apply_identity for each step of the filtration in order of
+   * the filtration. The bars of the diagram are retrieved via the given callback method every time
+   * a pair with non-zero length is closed. To retrieve the open/infinite bars, use @ref get_current_infinite_intervals.
+   *
+   * @param stream_interval Callback method to process the birth and death values of a persistence bar.
+   * Has to take three arguments as input: first the dimension of the cycle, then the birth value of the cycle
+   * and third the death value of the cycle. The values corresponds to the filtration values which were given at
+   * insertions or removals.
+   * @param minNumberOfFaces Minimum number of faces that will be in a complex at some point in the filtration.
+   * If the maximal number of faces is known in advance, the memory allocation can be better optimized.
+   * Default value: 0.
+   */
+  Filtered_zigzag_persistence(std::function<void(dimension_type, filtration_value, filtration_value)> stream_interval,
+                              unsigned int minNumberOfFaces = 0)
+      : handleToKey_(minNumberOfFaces),
+        numArrow_(-1),
+        keyToFiltrationValue_(minNumberOfFaces),
+        stream_interval_(std::move(stream_interval)),
```

Comment:
I completely forgot that you could stores arguments like that in lambdas (even though I knew that we put a `&` there because of it in the first place...), so I stored it out of reflex.